File: python/0049.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        ma = dict()
        for i in strs:
            k = ''.join(sorted(list(i.lower())))
            if k not in ma:
                ma[k] = []
            ma[k].append(i)
        ans = []
        for i in ma:
            ans.append(ma[i])
        return ans

File: rust/src/bin/0049.rs
struct Solution;

use std::collections::HashMap;
use std::iter::FromIterator;

impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut ma = HashMap::new();
        for i in &strs {
            let mut chars = i.chars().collect::<Vec<char>>();
            chars.sort();
            let k = String::from_iter(chars);
            if !ma.contains_key(&k) {
                ma.insert(k, vec!(i.clone()));
            } else {
                let ans = ma.get_mut(&k).unwrap();
                ans.push(i.clone());
            }
        }
        let mut ans = vec!();
        for i in ma.values() {
            ans.push(i.to_owned());
        }
        ans
    }
}

fn main() {
    println!("haha");
}